torch/_inductor/kernel/mm.py

```diff
@@ -89,7 +89,7 @@
             b = tl.load(B, mask=rk[:, None] < k, other=0.)
         if B_PROLOGUE_CAST_TYPE is not None:
             b = b.to(B_PROLOGUE_CAST_TYPE)
-        acc += tl.dot(a, b, allow_tf32=ALLOW_TF32)
+        acc += tl.dot(a, b, input_precision=INPUT_PRECISION)
```

Comment:
Moving the conversation here:
As reported by CUTLASS, this trick just gives higher precision for large enough matrices (k >= 200 or so). See https://www.nvidia.com/en-us/on-demand/session/gtcspring22-s41996/?playlistId=playList-d59c3dc3-9e5a-404d-8725-4b567f4dfe77 slide 39
![image](https://github.com/pytorch/pytorch/assets/3291265/47cfa455-2569-4112-a414-76e30cdc18b6)

I think we should perform a comprehensive benchmark of accuracy and also performance for smaller sizes if we are going to activate this by default.